app/lib/katello/resources/candlepin/consumer.rb

```diff
@@ -35,6 +35,15 @@ def get_all(uuids)
             consumers
           end
 
+          # workaround for https://bugzilla.redhat.com/1647724
```

Comment:
the number of hyprevisors won't be too big I think and it also runs asynchronously... we should ask QE for explicit test if we're in doubt, but this should be good until the candlepin provides data on /consumers endpoint

Reply:
We're currently batching the requests 75 at a time, so if we have 15k hypervisors (and users do), the old method makes 200 HTTPS requests, the new one 15000. I'll run a few tests on my local box to have a few before/after numbers.

Reply:
with 500 hypervisors, 50 guests each

before patch: 0(run)+110(finalize) sec new, 0(run)+20(finalize) sec update
after patch: 180(run)+200(finalize) sec new, 75(run)+175(finalize) sec update 

I'll see how we can speed that up :)

Reply:
@lzap @jlsherrill  if you have ideas, please :)

Reply:
My idea is to immediately add telemetry to this file to get some real-world numbers, then we can talk about poking the codebase. It's super easy the very same thing is done in this plugin: https://github.com/theforeman/foreman_discovery/pull/408/files#diff-45af6b2f1c078550eba223b206b16cb4

Reply:
@evgeni yeah, we used to query each host individually and recently changed to not do that to speed things up.  I'm not surprised this slows it down quite a bit.  

If we are okay with facts being imported eventually (but not necessarily immediately), one option would be to use our 'event_queue' to import them.  This would involve just throwing the host id on the queue, and then it would get processed in the background and would eventually be imported.